src/kitovu/sync/syncing.py

```diff
@@ -65,3 +73,19 @@ def start(connection_settings: ConnectionSettings) -> None:
             print(f'Local digest: {digest}')
 
     plugin.disconnect()
+
+
+def config_error(config_file: typing.Optional[pathlib.Path]) -> typing.Union[str, None]:
+    """Validates the given configuration file.
+
+    Returns either an error message or None if it's valid."""
+    try:
+        settings = Settings.from_yaml_file(config_file)
+        validator = utils.SchemaValidator(abort=False)
+        for _connection_key, connection_settings in sorted(settings.connections.items()):
+            _find_plugin(connection_settings, validator)
+        if validator.valid:
+            return None
+        return validator.error_message
+    except utils.UsageError as error:
```

Comment:
For example if the config file doesn't exist `from_yaml_file` throws an `UsageError`. Another source for a `UsageError` is `_find_plugin` which could throw a `NoPluginError`.

I could move the except three lines to the top, but I think (at least for me) that this would be more confusing. Otherwise I could make two `try` blocks, but that would lead to 3 more lines, which are IMHO unnecessary